zerver/actions/streams.py

```diff
@@ -1683,7 +1683,7 @@ def do_change_stream_group_based_setting(
         )
 
     setattr(stream, setting_name, user_group)
-    stream.save()
+    stream.save(update_fields=[setting_name, "name"])
```

Comment:
Why is `name` on this list? `stream.name` was not modified.

Reply:
It was modified in [test_rename_stream](https://github.com/zulip/zulip/blob/main/zerver/tests/test_subs.py#L2169), I tested that by printing the stream.name --fetched from db-- before and after `stream.save()`

Reply:
I don't think `name` is a field that `do_change_stream_group_based_setting` should be handling? And I don't see a test there that passes `name` as the `setting_name` parameter. If there is one, the test is bugged, and should call the function for changing a channel name, not `do_change_stream_group_based_setting`.

Reply:
`do_change_stream_group_based_setting` didn't change the name, it got changed in other code, There is a difference in names in my above test only because of  `stream.save()`.